gen/src/index.ts

```diff
@@ -37,42 +33,52 @@ async function main() {
   const templateDirectoryPath = `${__dirname}/../../${TEMPLATE_DIRECTORY}`;
 
   const schemasAndGenFiles = await qt.getJSONSchemasAndGenFiles(IN, LANGUAGE);
-  const allEvents: Event[] = [];
+  const allEventsByPkg: Map<string, Event[]> = new Map<string, Event[]>();
   schemasAndGenFiles.map(([schema, genFile]: [any, string]) => {
     // Write generated Python scripts
     const pkg = schema['$id'];
     const pkgPath = pkg.replace(/\./g, '/');
     mkdirp.sync(`${OUT}/${SRC_DIRECTORY}/${pkgPath}`);
+    const eventName = schema.name;
     fs.writeFileSync(
-      `${OUT}/${SRC_DIRECTORY}/${pkgPath}/${PY_FILENAME}`,
+      `${OUT}/${SRC_DIRECTORY}/${pkgPath}/${eventName}.py`,
       genFile
     );
 
-    const pkgEvents: string[] = [];
-    Object.keys(schema.properties).map((eventName: string) => {
-      const eventDescription = schema.properties[eventName].description;
-      pkgEvents.push(eventName);
-      allEvents.push({
-        package: pkg,
-        eventName: eventName,
-        eventDescription: eventDescription,
-      });
-    });
+    const eventDescription = schema.description;
+    const event = {
+      package: pkg,
+      eventName: eventName,
+      eventDescription: eventDescription,
+    };
 
-    // Write __init__.py scripts
-    const initPySqrlTmpl = fs.readFileSync(
-      `${templateDirectoryPath}/${INIT_PY_TEMPLATE}`
-    );
+    if (allEventsByPkg.has(pkg)) {
+      const pkgEvents = allEventsByPkg.get(pkg);
+      pkgEvents?.push(event);
+    } else {
+      const pkgEvents: Event[] = [event];
+      allEventsByPkg.set(pkg, pkgEvents);
+    }
+  });
+
+  // Write __init__.py scripts
+  const initPySqrlTmpl = fs.readFileSync(
+    `${templateDirectoryPath}/${INIT_PY_TEMPLATE}`
+  );
+  for (const k of allEventsByPkg.keys()) {
```

Comment:
Spell out keys. Or even better, spell out what the the keys represent.

`packageKey`?

Reply:
Done.